### PR TITLE
add validator to range param

### PR DESCRIPTION
**Pull request by azazellochg**
_getListFromRangeString from pw.viewer should be removed as a duplicate of pwutils.getListFromRangeString

### CODE REVIEW

File: pyworkflow/object.py [hunk 1]
```diff
@@ -798,9 +798,33 @@ def __nonzero__(self):
         return self.get() 
     
     def __bool__(self):
-        return self.get()  
-    
-    
+        return self.get()
+
+
+class Range(String):
```
Comment (delarosatrevin) [inline pyworkflow/object.py:804]: I don't see the need for this new class

Comment (pconesa) [reply pyworkflow/object.py:804]: Wouldn't this turn user input "1,5-8,10" into [1,5,6,7,8,10] in the form next time it loads the protocol?

File: pyworkflow/protocol/params.py [hunk 2]
```diff
@@ -548,8 +548,8 @@ class NumericRangeParam(StringParam):
 
     """
     def __init__(self, **args):
-        StringParam.__init__(self, **args)
-        # TODO: ADD a syntax validator
+        Param.__init__(self, paramClass=Range, **args)
+        self.addValidator(NumericRangeValidator())
```
Comment (pconesa) [inline pyworkflow/protocol/params.py:552]: Yeah! one TODO less! Nice

**Comment by azazellochg** [comment]
Hi @pconesa I will not have time to try this new methods, please ask someone else to review the PR

**Review by pconesa — APPROVED** [review]

**Pull request merged**